components/core/src/clp_s/timestamp_parser/TimestampParser.cpp

```diff
@@ -1207,6 +1216,26 @@ auto parse_timestamp(
                 timestamp_idx += cFieldLength;
                 break;
             }
+            case 'J': {  // Leap second.
+                constexpr size_t cFieldLength{2};
+                if (timestamp_idx + cFieldLength > timestamp.size()) {
+                    return ErrorCode{ErrorCodeEnum::IncompatibleTimestampPattern};
+                }
+
+                auto const parsed_leap_second{
+                        YSTDLIB_ERROR_HANDLING_TRYX(convert_padded_string_to_number(
+                                timestamp.substr(timestamp_idx, cFieldLength),
+                                '0'
+                        ))
+                };
+                if (cParsedLeapSecond != parsed_leap_second) {
+                    return ErrorCode{ErrorCodeEnum::IncompatibleTimestampPattern};
```

Comment:
```suggestion
                if (cParsedLeapSecond
                    != YSTDLIB_ERROR_HANDLING_TRYX(convert_padded_string_to_number(
                            timestamp.substr(timestamp_idx, cFieldLength),
                            '0'
                    )))
                {
```
Nit: I feel like it's more clear if we don't have both `parsed_leap_second` and `parsed_second`.